### PR TITLE
LinalgRefactor - Enable matrix block elementwise product

### PR DESCRIPTION
**Pull request by OXPHOS**
to replace old `linalg::elementwise_square`

### CODE REVIEW

File: src/shogun/mathematics/linalg/LinalgNamespace.h [hunk 1]
```diff
@@ -309,6 +309,56 @@ T dot(const SGVector<T>& a, const SGVector<T>& b)
 	return infer_backend(a, b)->dot(a, b);
 }
 
+/** Performs the operation C = A .* B where ".*" denotes elementwise multiplication
+ * on matrix blocks.
+ *
+ * This version returns the result in-place.
+ * User should pass an appropriately allocate memory matrix
+ * Or pass one of the operands arguments (A or B) as a result
```
Comment (karlnapf) [inline src/shogun/mathematics/linalg/LinalgNamespace.h]: Watch your English here :) Punctuation, grammar etc.
Super minor!

**Comment by karlnapf** [comment]
Merge conflicts, otherwise good to go!

**Comment by OXPHOS** [comment]
@karlnapf ready

**Comment by karlnapf** [comment]
Travis error unrelated, merging

**Pull request merged**